src/Voron/Impl/Journal/WriteAheadJournal.cs

```diff
@@ -2125,7 +1903,7 @@ private void ReduceSizeOfCompressionBufferIfNeeded(bool forceReduce = false)
                 // PERF: Compression buffer will be reused, it is safe to discard the content to clear the modified bit.
                 // For encrypted databases, discarding locked memory is *expensive*, so we avoid it
                 if (_env.Options.Encryption.IsEnabled == false)
-                    _compressionPager.DiscardWholeFile();
+                    _compressionPager.DiscardWholeFile(_compressionPagerState);
 
```

Comment:
Cant we push this onto the Pager itself?

Reply:
No, we _explicitly_ not holding the pager state in the pager.

Reply:
What I mean is this decision:
` if (_env.Options.Encryption.IsEnabled == false)`

Reply:
That isn't a property of the pager, however. This is a property of the storage environment.
In particular, this specific pager isn't even encrypted (we just lock the data in memory)